### PR TITLE
Feature/vg 9445 generic parser erc721 erc1155

### PR DESCRIPTION
**Pull request by mgrattepanche-ledger**
This PR is about addind generic ERC721 and ERC1155 parsers

### CODE REVIEW

File: ethereum/parsers.schema.json [hunk 1]
```diff
@@ -174,7 +174,10 @@
                                         "claim_reward",
                                         "configure",
                                         "buy",
-                                        "approve"
+                                        "approve",
+                                        "read",
```
Comment (slundy-ledger) [inline ethereum/parsers.schema.json:178]: "read" is not present in PR adding new function type in PSM
https://github.com/LedgerHQ/python-smart-contract/pull/56/files

**Review by slundy-ledger — COMMENTED** [review]
what about splitting the `erc` folder into `erc721` and `erc1155` ? for better visibility of what it is

File: ethereum/erc/parsers.json [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                                "uint256": {
+                                    "path": {
+                                        "default": "tokenId"
+                                    }
+                                }
+                            }
+                        }
+                    ],
+                    "functionName": "safeTransferFrom",
+                    "functionType": "transfer721",
```
Comment (slundy-ledger) [inline ethereum/erc/parsers.json:46]: ERROR:validation.validators:Validation error: {'file': 'ethereum/erc/parsers.json', 'message': "'transfer721' is not one of ['stake', 'swap', 'wrap', 'unwrap', 'borrow', 'repay_loan', 'lend', 'withdraw', 'claim_reward', 'configure', 'buy', 'approve', 'read', 'transfer_erc721', 'transfer_erc1155']"}

File: ethereum/erc/parsers.json [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                                "uint256": {
+                                    "path": {
+                                        "default": "amount"
+                                    }
+                                }
+                            }
+                        }
+                    ],
+                    "functionName": "safeTransferFrom",
+                    "functionType": "transfer1155",
```
Comment (slundy-ledger) [inline ethereum/erc/parsers.json:118]: transfer1155' is not one of ['stake', 'swap', 'wrap', 'unwrap', 'borrow', 'repay_loan', 'lend', 'withdraw', 'claim_reward', 'configure', 'buy', 'approve', 'read', 'transfer_erc721', 'transfer_erc1155']

**Comment by mgrattepanche-ledger** [comment]
Im not really in favor of splitting in disctinct parsers cause we'll have to handle multiple `dapp_name` and `dapp_id`.
As they are all related to NFTs, would be nice to have a way to keep them gathered under parsers.

**Comment by mgrattepanche-ledger** [comment]
:arrow_up: Misclick :eyes:

**Comment by slundy-ledger** [comment]
> Im not really in favor of splitting in disctinct parsers cause we'll have to handle multiple `dapp_name` and `dapp_id`. As they are all related to NFTs, would be nice to have a way to keep them gathered under parsers.

Then what about erc_token instead of erc ?

**Review by slundy-ledger — APPROVED** [review]

**Pull request merged**